binaries/speed_benchmark_torch.cc

```diff
@@ -129,6 +138,24 @@ int main(int argc, char** argv) {
   torch::jit::GraphOptimizerEnabledGuard no_optimizer_guard(false);
   auto module = torch::jit::load(FLAGS_model);
 
+  if (FLAGS_use_bundled_input >= 0) {
+    auto get_method = module.find_method("get_all_bundled_inputs");
+    if (!get_method) {
+      std::cerr << "Model does not have bundled inputs.  Before saving," << std::endl
+        << "use torch.utils.bundled_inputs.augment_model_with_bundled_inputs." << std::endl;
+      return 1;
+    }
+
+    auto all_inputs = (*get_method)({}).toList();
+    if (FLAGS_use_bundled_input >= all_inputs.size()) {
+      // NOTE: This check is only to make the error message nicer.
+      // The get call below does internal bounds checking.
+      std::cerr << "Model has only " << all_inputs.size() << " bundled inputs." << std::endl;
+      return 1;
+    }
+    inputs = all_inputs.get(FLAGS_use_bundled_input).toTuple()->elements();
+  }
+
   module.eval();
   if (FLAGS_print_output) {
     std::cout << module.forward(inputs) << std::endl;
```

Comment:
Why not just call `run_on_bundled_input`?

Reply:
I thought about this, since that's what `run_on_bundled_input` was designed for.  I decided against it for two reasons.

1. This way is a less invasive change to the code.
1. In the real world, passing the arguments from C++ is part of the run time of the model, so we should include that in the profile.